shadow.c

```diff
@@ -117,10 +118,19 @@ static PHP_GSHUTDOWN_FUNCTION(shadow)
 }
 /* }}} */
 
+/* {{{ shadow_deps
+ */
+static const zend_module_dep shadow_deps[] = {
+	ZEND_MOD_OPTIONAL("zip")
```

Comment:
It still looks unfair to me that we list only some of the extensions here. E.g. GD is usually compiled as a shared extension and mostly interacts with files.

This approach in some cases can make debugging even harder. Say, the extensions are loaded as `shadow` → `gd` → `zip`. In this case the `zip` functions will be overridden, but the `gd`'s won't, and it won't be clear that that's because of the load order.